src/test/scala/org/ergoplatform/mining/ErgoMinerSpec.scala

```diff
@@ -61,6 +71,90 @@ class ErgoMinerSpec extends TestKit(ActorSystem()) with FlatSpecLike with ErgoTe
+    val (l, r) = inputs.splitAt(50)
+    val tx_1 = validTransactionGen(l).sample.get
+    val tx_2 = validTransactionGen(r :+ l.last).sample.get
+
+    ErgoMiner.fixTxsConflicts(Seq(tx_1, tx_2, tx)) should contain theSameElementsAs Seq(tx_1, tx)
+    ErgoMiner.fixTxsConflicts(Seq(tx_2, tx_1, tx)) should contain theSameElementsAs Seq(tx_2, tx)
+  }
+
+  ignore should "create own coinbase transaction, if there is already a transaction, that spends emission box" in {
+    // TODO
+  }
+
+  it should "work correctly with 2 coinbase txs in pool" in {
+    val tmpDir = createTempDir
+
+    type msgType = SemanticallySuccessfulModifier[_]
+    val msgClass = classOf[msgType]
+    val testProbe = new TestProbe(system)
+    system.eventStream.subscribe(testProbe.ref, msgClass)
+
+    val defaultSettings: ErgoSettings = ErgoSettings.read(None).copy(directory = tmpDir.getAbsolutePath)
+
+    val nodeSettings = defaultSettings.nodeSettings.copy(mining = true,
+      stateType = StateType.Utxo,
+      miningDelay = defaultAwaitDuration,
+      offlineGeneration = true,
+      verifyTransactions = true)
+    val chainSettings = defaultSettings.chainSettings.copy(blockInterval = 2.seconds)
+    val ergoSettings = defaultSettings.copy(nodeSettings = nodeSettings, chainSettings = chainSettings)
+
+    val nodeViewHolderRef: ActorRef = ErgoNodeViewRef(ergoSettings, timeProvider, emission)
+    val readersHolderRef: ActorRef = ErgoReadersHolderRef(nodeViewHolderRef)
+
+    val minerRef: ActorRef = ErgoMinerRef(ergoSettings, nodeViewHolderRef, readersHolderRef, timeProvider, emission, TrueLeaf)
+    expectNoMessage(1 second)
+    val r: Readers = Await.result((readersHolderRef ? GetReaders).mapTo[Readers], 10 seconds)
+
+    val state: UtxoStateReader = r.s.asInstanceOf[UtxoStateReader]
+    val history: ErgoHistoryReader = r.h
+    val startBlock = history.bestHeaderOpt
+
+    val newBLockDuration = 30 seconds
+
+    minerRef ! StartMining
+
+    testProbe.expectMsgClass(newBLockDuration, msgClass)
+
+    val prop1 = DLogProverInput(BigIntegers.fromUnsignedByteArray("test1".getBytes())).publicImage
+    val prop2 = DLogProverInput(BigIntegers.fromUnsignedByteArray("test2".getBytes())).publicImage
+
+    val boxToDoubleSpend = r.h.bestFullBlockOpt.get.transactions.last.outputs.last
+    val input = Input(boxToDoubleSpend.id, ProverResult(Array.emptyByteArray, ContextExtension.empty))
+
+    val outputs1 = IndexedSeq(new ErgoBoxCandidate(boxToDoubleSpend.value, prop1))
+    val tx1 = new ErgoTransaction(IndexedSeq(input), outputs1)
+
+    val outputs2 = IndexedSeq(new ErgoBoxCandidate(boxToDoubleSpend.value, prop2))
+    val tx2 = new ErgoTransaction(IndexedSeq(input), outputs2)
+
+    nodeViewHolderRef ! LocallyGeneratedTransaction[ErgoTransaction](tx1)
+    nodeViewHolderRef ! LocallyGeneratedTransaction[ErgoTransaction](tx2)
+    expectNoMessage(1 seconds)
+
+    r.m.unconfirmed.size shouldBe 2
+
+    testProbe.expectMsgClass(newBLockDuration, msgClass)
```

Comment:
Why waiting for 3 blocks here?

Reply:
@catena2w cause miner already have new candidates while we are making this tx, 3 blocks is `for sure` duration for tx to be mined.